Source/Core/DolphinQt/Config/GameConfigHighlighter.h

```diff
@@ -0,0 +1,29 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QRegularExpression>
+#include <QSyntaxHighlighter>
+#include <QTextCharFormat>
+
+class GameConfigHighlighter : public QSyntaxHighlighter
+{
+  Q_OBJECT
+
+public:
+  GameConfigHighlighter(QTextDocument* parent = nullptr);
+
+protected:
+  void highlightBlock(const QString& text) override;
+
+private:
+  struct HighlightingRule
+  {
+    QRegularExpression pattern;
+    QTextCharFormat format;
+  };
+
+  QVector<HighlightingRule> m_rules;
```

Comment:
with `std::vector` `HighlightingRule` can be forward declared, and the relevant includes can be moved to the cpp file (destructor will need to be defaulted in the cpp file as well).